### ISSUE TITLE
Obtain similar performance with BE data with supervised training

### ISSUE THREAD

**Issue by taeil** [Berkeley-Data/irrigation_detection#35]
Training with up to 50% dataset and 50 epochs (no further tuning) does not show the result that previous team presented. 

- either missing some critical steps 
- missing critical information for dataset 

https://wandb.ai/taeil/irrigation_detection/reports/Supervised-Training-With-BigEarthNet--Vmlldzo0NTg5NTQ

Additionally, 
- augmentation code is not working 
- dataset is a little confusing in terms of labels. 
- currently it seems to be running multi-class detection (but it is not clear on the code) 

Here is tutorial for TFRecord for Keras https://www.tensorflow.org/datasets/keras_example

**Comment by taeil** [comment]
I think I finally cracked the code. The secret is in the data loading.
utils.py
def read_tfrecord(example):
...
...
example['binary_label'] = example['original_labels_multi_hot'][tf.constant(12)]
...
...
"Permanently irrigated land": 12
Therefore, this is binary classification.

**Comment by taeil** [comment]
How to split data. 
- https://stackoverflow.com/questions/51125266/how-do-i-split-tensorflow-datasets/58452268#58452268 
- https://stackoverflow.com/questions/48213766/split-a-dataset-created-by-tensorflow-dataset-api-in-to-train-and-test

**Comment by taeil** [comment]
[Downsampling and Upweighting](https://developers.google.com/machine-learning/data-prep/construct/sampling-splitting/imbalanced-data)
